tachyon/c/crypto/commitments/fri/two_adic_fri_impl.h

```diff
@@ -22,39 +22,32 @@ class TwoAdicFRIImpl
   using Commitment = typename Base::Commitment;
   using ProverData = typename Base::ProverData;
   using Domain = typename Base::Domain;
-  using OpeningPoints = typename Base::OpeningPoints;
-  using OpenedValues = typename Base::OpenedValues;
-  using FRIProof = typename Base::FRIProof;
 
   using Base::Base;
 
   void AllocateLDEs(size_t size) { ldes_.reserve(size); }
 
-  template <typename Derived>
-  absl::Span<F> CosetLDEBatch(Eigen::MatrixBase<Derived>& matrix, F shift) {
+  void CosetLDEBatch(Eigen::Map<tachyon::math::RowMajorMatrix<F>>& matrix,
+                     F shift,
+                     Eigen::Map<tachyon::math::RowMajorMatrix<F>>& lde) {
```

Comment:
Since `Eigen::Map` has only pointer,
```suggestion
                     Eigen::Map<tachyon::math::RowMajorMatrix<F>> lde) {
```